### ISSUE TITLE
Push 2 - Show play head in Session mode

### ISSUE THREAD

**Issue by saponace** [git-moss/DrivenByMoss#268]
Hi,

First of all, thank you very much for your amazing work. I have been using the push2 on Bitwig for more than a year thanks to you !

There is a feature that I think would be great to have: Being able to see the play head on the push2 screen in session mode.

Playing clips (more than one bar) can be challenging since the user can not see on the push screen where the playing clip is at, and hence can't queue other clips play accordingly.

For reference, Bitwig's clip launcher shows said play head of each playing clip.

Do you think this could be doable ?

**Comment by git-moss** [comment]
That is on my wishlist, too but sadly so far not possible.

**Comment by saponace** [comment]
Just for my understanding, is it a limitation in Bitwig API ?

**Comment by git-moss** [comment]
Yes.

**Comment by saponace** [comment]
Hey,

So I dug a little in your code in order to understand a bit what's going on, and found the following. Can you confirm this is correct ?

In order to display the play head (and midi notes too, that would be cool) in the clips in Session mode, we need  the following data:
* Clip length: Is there a way to get this ? I saw you initialize `INoteClips` with a length of 8 instead of the actual length of a clip, so I guess not
* Position of the play head (Current playing step in this clip): Accessible through `Clip.playingStep()`. But only one `Clip` is accessible at a time (the selected clip in Bitwig).

So basically we are waiting for Bitwig to add the ability to get the inner `Clip` of a `ClipLauncherSlot`, and the length of a `Clip` ?

**Comment by git-moss** [comment]
Sorry, I cannot support you with this. The specification is available publicly from Ableton: https://github.com/Ableton/push-interface
